### PR TITLE
update migration guide for messaging

### PR DESCRIPTION
**Pull request by yiliuTo**
as subject

### CODE REVIEW

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 1]
```diff
@@ -762,6 +790,52 @@ public class Demo {
...
+* For the batch consuming mode, change the message header names converted from batched messages.
+    - Change message header from `azure_eventhub_enqueued_time` to `azure_eventhub_batch_converted_enqueued_time`.
+    - Change message header from `azure_eventhub_offset` to `azure_eventhub_batch_converted_offset`.
+    - Change message header from `azure_eventhub_sequence_number` to `azure_eventhub_batch_converted_sequence_number`.
+    - Change message header from `azure_partition_key` to `azure_batch_converted_partition_key`.
+* When publishing messages to Event Hubs, ignore all message headers converted from batched messages. Headers include:
+    - azure_eventhub_batch_converted_enqueued_time
+    - azure_eventhub_batch_converted_offset
+    - azure_eventhub_batch_converted_sequence_number
+    - azure_batch_converted_partition_key
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:821]: same for this one, put it as the first one

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 2]
```diff
@@ -777,55 +851,50 @@ version 2 of link:https://search.maven.org/artifact/com.azure.spring/azure-sprin
...
+* Drop API `setClientConfig` of `ServiceBusQueueOperation` and `ServiceBusTopicOperation`. To configure the underlying `ServiceBusProcessorClient` used by the inbound channel adapter, users can use `ServiceBusContainerProperties` instead.
+* Drop `CompletableFuture` support of `ServiceBusTemplate` and `DefaultMessageHandler` and support Reactor instead.
+* Add new API of `setDefaultEntityType` for `ServiceBusTemplate` to specify the entity type, which is required when no bean of `PropertiesSupplier&lt;String, ProducerProperties&gt;` is provided for the `ProducerProperties#entityType`.
+* Drop message header of `AzureHeaders.RAW_ID`. Please use `ServiceBusMessageHeaders.MESSAGE_ID` instead.
 
 .Class mapping from azure-spring-cloud-starter-servicebus to spring-cloud-azure-starter-integration-servicebus
 [cols="<,<,<", options="header"]
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:882]: two columns

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 3]
```diff
@@ -904,6 +1018,38 @@ public class Demo {
...
+* Drop API `setCheckpointConfig` of ServiceBusTopicOperation or ServiceBusQueueOperation. To set the checkpoint configuration for the inbound channel adapter, users can call the method `ServiceBusContainerProperties#setAutoComplete` instead. To disable the auto-complete mode is equivalent to `MANUAL` checkpoint mode and to enable it will trigger the `RECORD` mode.
+* Drop API `setClientConfig` of `ServiceBusQueueOperation` and `ServiceBusTopicOperation`. To configure the underlying `ServiceBusProcessorClient` used by the inbound channel adapter, users can use `ServiceBusContainerProperties` instead.
+* Drop `CompletableFuture` support of `ServiceBusTemplate` and `DefaultMessageHandler` and support Reactor instead.
+* Add new API of `setDefaultEntityType` for `ServiceBusTemplate` to specify the entity type, which is required when no bean of `PropertiesSupplier&lt;String, ProducerProperties&gt;` is provided for the `ProducerProperties#entityType`.
+* Drop message header of `AzureHeaders.RAW_ID`. Please use `ServiceBusMessageHeaders.MESSAGE_ID` instead.
+
+.Class mapping from azure-spring-cloud-starter-servicebus to spring-cloud-azure-starter-integration-servicebus
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1041]: this table should be called `Class mapping from azure-spring-integration-servicebus to spring-integration-azure-servicebus`

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 4]
```diff
@@ -933,26 +1079,46 @@ All configuration property names changed the prefix from `spring.cloud.azure.sto
...
+* Please refer to <<migration-guide-naming, naming changes>> to learn more about the strategy changes in the project naming.
+* Please refer to <<migration-guide-bom, bom>> to learn how to using one BOM for all Spring Cloud Azure libraries.
+
+[#api-spring-integration-azure-storage-queue]
+===== API Changes
+
+* Drop `StorageQueueOperation` and provide `StorageQueueTemplate` instead.
+* Do not support configuration of checkpoint mode in `StorageQueueTemplate`, only the `MANUAL` mode is supported.
+
+.Class mapping from azure-spring-cloud-starter-storage-queue to spring-cloud-azure-starter-integration-storage-queue
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1112]: same here from `azure-spring-integration-storage-queue`

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 5]
```diff
@@ -1082,37 +1251,40 @@ or link:https://search.maven.org/artifact/com.azure.spring/azure-spring-cloud-st
 
 [#configuration-spring-cloud-azure-stream-binder-servicebus]
 ===== SDK Configuration changes
-IMPORTANT: Legacy binder libaries are `azure-spring-cloud-stream-binder-servicebus-queue` and `azure-spring-cloud-stream-binder-servicebus-topic`, and now they're merged into one `spring-cloud-azure-stream-binder-servicebus`.
+IMPORTANT: Legacy binder libraries are `azure-spring-cloud-stream-binder-servicebus-queue` and `azure-spring-cloud-stream-binder-servicebus-topic`, and now they're merged into one `spring-cloud-azure-stream-binder-servicebus`
+to support both Service Bus Queue and Topic.
 
-We've merged these two libraries into one, which supports both topic and queue. And the binder type is combined as `servicebus`.
+IMPORTANT: tnd the binder type is combined from `servicebus-queue` and `servicebus-topic` as `servicebus`.
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1257]: `tnd`

**Review by saragluna — CHANGES_REQUESTED** [review]

**Comment by saragluna** [comment]
One more thing we should mention specifically is that for spring cloud libraries, they used to use the service principal to authenticate with ARM and then retrieve the connection string, we should point out:
- How they can achieve the same thing and what are the properties they should configure.
- Other options instead of using the ARM.
This is not required in this PR and we can add this later.

**Comment by yiliuTo** [comment]
Good idea, I will create another pr for this.

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**